src/transforms/ansi_stripper.rs

```diff
@@ -0,0 +1,127 @@
+use super::Transform;
+use crate::{
+    event::{self, ValueKind},
+    topology::config::{DataType, TransformConfig, TransformDescription},
+    Event,
+};
+use serde::{Deserialize, Serialize};
+use string_cache::DefaultAtom as Atom;
+
+#[derive(Deserialize, Serialize, Debug)]
+#[serde(deny_unknown_fields)]
+pub struct AnsiStripperConfig {
+    field: Option<Atom>,
+}
+
+inventory::submit! {
+    TransformDescription::new_without_default::<AnsiStripperConfig>("ansi_stripper")
+}
+
+#[typetag::serde(name = "ansi_stripper")]
+impl TransformConfig for AnsiStripperConfig {
+    fn build(&self) -> crate::Result<Box<dyn Transform>> {
+        let field = self.field.as_ref().unwrap_or(&event::MESSAGE);
+
+        Ok(Box::new(AnsiStripper {
+            field: field.clone(),
+        }))
+    }
+
+    fn input_type(&self) -> DataType {
+        DataType::Log
+    }
+
+    fn output_type(&self) -> DataType {
+        DataType::Log
+    }
+
+    fn transform_type(&self) -> &'static str {
+        "ansi_stripper"
+    }
+}
+
+pub struct AnsiStripper {
+    field: Atom,
+}
+
+impl Transform for AnsiStripper {
+    fn transform(&mut self, mut event: Event) -> Option<Event> {
+        let log = event.as_mut_log();
+
+        match log.get_mut(&self.field) {
+            None => debug!(
+                message = "Field does not exist.",
+                field = self.field.as_ref(),
+            ),
+            Some(ValueKind::Bytes(ref mut bytes)) => {
+                *bytes = match strip_ansi_escapes::strip(bytes.clone()) {
+                    Ok(b) => b.into(),
+                    Err(error) => {
+                        debug!(
+                            message = "Could not strip ANSI escape sequences.",
+                            field = self.field.as_ref(),
+                            %error,
+                            rate_limit_secs = 30,
+                        );
+                        return Some(event);
+                    }
+                };
+            }
+            _ => debug!(
+                message = "Field value must be a string.",
+                field = self.field.as_ref(),
+            ),
+        }
+
+        Some(event)
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::AnsiStripper;
+    use crate::{
+        event::{Event, ValueKind, MESSAGE},
+        transforms::Transform,
+    };
+
+    macro_rules! assert_foo_bar {
+        ($($in:expr),* $(,)?) => {
+            $(
+                let mut transform = AnsiStripper {
+                    field: "message".into(),
+                };
+
+                let event = Event::from($in);
+                let event = transform.transform(event).unwrap();
+
+                assert_eq!(
+                    event.into_log().into_value(&MESSAGE).unwrap(),
+                    ValueKind::from("foo bar")
+                );
+            )+
+        };
+    }
+
+    #[test]
+    fn ansi_stripper_transform() {
+        assert_foo_bar![
```

Comment:
This tests all potential escape sequences as described in http://ascii-table.com/ansi-escape-sequences.php.